futhark_ffi/__init__.py

```diff
@@ -118,8 +118,8 @@ def make_wrapper(self, ff):
         @wraps(ff)
         def wrapper(*args):
             out_args = [self.ffi.new(t) for t in out_types]
-            ff(self.ctx, *out_args,
-               *(f(a) for f, a in zip(converters, args)))
+            in_args = [f(a) for f, a in zip(converters, args)]
+            ff(self.ctx, *(out_args+in_args))
```

Comment:
Does 2.7 not support multiple varargs? `*out_args, *in_args` is probably faster than first concatenating them.

Reply:
No, I got a parse error when I tried that.  I have no idea why it isn't supported.